cloud/spawn.go

```diff
@@ -324,15 +324,10 @@ func ModifySpawnHost(ctx context.Context, env evergreen.Environment, host *host.
 func MakeExtendedSpawnHostExpiration(host *host.Host, extendBy time.Duration) (time.Time, error) {
 	if err := host.PastMaxExpiration(extendBy); err != nil {
 		return time.Time{}, err
+	} else {
+		newExp := host.ExpirationTime.Add(extendBy)
+		return newExp, nil
 	}
-
-	newExp := host.ExpirationTime.Add(extendBy)
-	remainingDuration := newExp.Sub(time.Now()) //nolint
-	if remainingDuration > evergreen.MaxSpawnHostExpirationDurationHours {
-		return time.Time{}, errors.Errorf("Can not extend host '%s' expiration by '%s'. Maximum host duration is limited to %s", host.Id, extendBy.String(), evergreen.MaxSpawnHostExpirationDurationHours.String())
```

Comment:
should we maybe move this check above `host.ExpirationTime.Add` instead of removing it?

Reply:
im not sure if we want to have the arbitrary 2 week extension maximum and the PastMaxExpiration should be checking everything (i think)

Reply:
Whether or not we should have an arbitrary 2 week extension maximum feels more like a product question. I think we should double check with Annie before removing it.

Reply:
The idea behind the 2 week extension check was just to prevent users from extending too far when they don't need to; that being said since we _do_ have the one month max so there's really no reason to have two levels of prevention (this was pre-nonexpirable hosts days when we were likely trying to be careful about money here). I'm fine with removing it.